src/gui/sdlmain.cpp

```diff
@@ -2229,6 +2241,71 @@ static SDL_Point window_bounds_from_label(const std::string &pref,
 	return {w, h};
 }
 
+static SDL_Point clamp_to_minimum_window_dimensions(SDL_Point size)
+{
+	const auto w = std::max(size.x, FALLBACK_WINDOW_DIMENSIONS.x);
+	const auto h = std::max(size.y, FALLBACK_WINDOW_DIMENSIONS.y);
+	return {w, h};
+}
+
+static SDL_Rect get_desktop_resolution()
+{
+	SDL_Rect desktop;
+	assert(sdl.display_number >= 0);
+	SDL_GetDisplayBounds(sdl.display_number, &desktop);
+	assert(desktop.w >= FALLBACK_WINDOW_DIMENSIONS.x);
+	assert(desktop.h >= FALLBACK_WINDOW_DIMENSIONS.y);
+	return desktop;
+}
+
+// Takes in:
+//  - The 'maxresolution' config value: 'default', 'W,H', or an invalid setting.
+//
+// Except for SURFACE and TEXTURE rendering, the function populates the following struct members:
+//  - 'sdl.desktop.restrict_resolution', true if resolution restriction is enabled.
+//  - 'sdl.desktop.max_resolution', with the refined size.
+
+static void setup_resolution_restriction_from_conf(const char *max_resolution_val)
+{
+	// TODO: Deprecate SURFACE output and remove this.
+	if (sdl.desktop.want_type == SCREEN_SURFACE)
+		return;
+
+	// TODO: Implement support for TEXTURE output.
+	if (sdl.desktop.want_type == SCREEN_TEXTURE)
+		return;
+
+	const auto desktop = get_desktop_resolution();
+	int w, h;
+	auto was_parsed = sscanf(max_resolution_val, "%dx%d", &w, &h) == 2;
```

Comment:
Suggest an immediate ballout here if it's not parsed.
```c++
if (!was_parsed) {
   LOG_WARNING(.. consequences..);
   return;
}
```

Reply:
Then subsequence checks for `was_parsed` can be dropped (beyond this point we're good for parsing).

Reply:
That's a better way of doing it, agreed.